### ISSUE TITLE
Kid Icarus Uprising - Hangs after saving

### ISSUE THREAD

**Issue by JohnGodgames** [citra-emu/citra#2238]
Tested on https://github.com/citra-emu/citra/commit/cf5b8483fc53ca3b4919d231045348625a507d05 .

First boot [LOG](http://pastebin.com/9ZJ67fTv)

bottom screen is clickable:
![kid 1](https://cloud.githubusercontent.com/assets/11453507/20746658/f8ba6e58-b6e6-11e6-9cf7-79509cd00ff8.PNG)

Second boot [LOG](http://pastebin.com/MqP7ji78)

screen:
![kid 2](https://cloud.githubusercontent.com/assets/11453507/20746662/fdd03a08-b6e6-11e6-8125-7fdce35d989e.PNG)

After saving spams with:
```
[ 138.420488] Service <Error> core\hle\service\service.cpp:Service::Interface::SyncRequest:74: unknown / unimplemented function 'ReadSavedData': port=cecd:u, cmd_buff[1]=0x60, cmd_buff[2]=0x30000, cmd_buff[3]=0x2, cmd_buff[4]=0x12, cmd_buff[5]=0x20, cmd_buff[6]=0x0, cmd_buff[7]=0x60C, cmd_buff[8]=0xE014D28
[ 138.420979] Service.NDM <Warning> core\hle\service\ndm\ndm.cpp:Service::NDM::ResumeDaemons:102: (STUBBED) daemon_bit_mask=0x0000000F
```

**Comment by Subv** [comment]
Another game that loops trying to access the CECD savedata, just like Tomodachi Life

**Comment by mailwl** [comment]
to pass this loop, just stub ReadSavedData, which return error (-1) for example

**Comment by SigmaVirus** [comment]
@mailwl What do you mean by stubbing ReadSavedData?

I can't get this game to work at all and experiencing the same as @JohnGodgames 

I have tried the latest nightly build of Citra and the bleeding-edge-190 build and it's still the same.

Have seen videos on YouTube demonstrating this game on Citra so it must work somehow.

**Comment by SigmaVirus** [comment]
OK, I created some pull requests for this (#2535, #2526 and #2537)
Sorry for the clutter, but I didn't know how to change multiple files at once.

**Comment by SigmaVirus** [comment]
Created a proper pull request for this #2538 according to what @mailwl said about stubbing.

**Comment by mailwl** [comment]
i dislike workaround in codebase, @wwylele what do you think about it? maybe as temporary hack is allowed,

**Comment by SigmaVirus** [comment]
I'd be happy just to see something else in the game other than the spinning loading logotype right now.
This probably is just a hack workaround like @mailwl says but right now, I'd be happy to play the game.

**Comment by SigmaVirus** [comment]
@JohnGodgames 
According to this [YouTube](https://www.youtube.com/watch?v=KqseBWgRuH8) video you've made you got the game running and from the looks of it I would say it looks really playable.

Care to share how you did it?

**Comment by JohnGodgames** [comment]
https://github.com/mailwl/citra/tree/kid

**Comment by mailwl** [comment]
yesm the game is really playable

**Comment by SigmaVirus** [comment]
@JohnGodgames 
Thanks, I managed to snag a copy of one of your builds from your FaceBook page and got the game to run at least.

Even with the specs I have (i7 4790K at stock, ASUS Nvdidia 1060 Strix OC and 16GB RAM) the game chugs at the "Land" sections of the chapters, "Sky" sections runs OK though however the game crashed for me at chapter 4 midstage.

The game seems to be really playable actually but again really slow at "Land" sections.

**Comment by andwhyisit** [comment]
@SigmaVirus: Mind sharing said build? I can't find it.

**Comment by SigmaVirus** [comment]
@andwhyisit 
[This](https://ci.appveyor.com/api/buildjobs/64ja1813ey8f6hv2/artifacts/citra-windows-mingw-20171113-6584df8.zip) should work and it's a recent build too!

**Comment by einstein95** [comment]
As of , the log shows
```
[   5.251035] Service.FS <Error> core/hle/service/fs/fs_user.cpp:OpenArchive:469: failed to get a handle for archive archive_id=0x00000006 archive_path=[Binary: 010000000003000000000000]
[   5.685263] Service.NDM <Warning> core/hle/service/ndm/ndm.cpp:SuspendDaemons:89: (STUBBED) daemon_bit_mask=0x0000000E
[   5.692643] Service <Error> core/hle/service/service.cpp:HandleSyncRequest:97: unknown / unimplemented function 'OpenAndRead': port=cecd:u, cmd_buff[1]=0x60, cmd_buff[2]=0x30000, cmd_buff[3]=0x2, cmd_buff[4]=0x12, cmd_buff[5]=0x20, cmd_buff[6]=0x0, cmd_buff[7]=0x60C, cmd_buff[8]=0xE014D28
[   5.696997] Service.NDM <Warning> core/hle/service/ndm/ndm.cpp:ResumeDaemons:104: (STUBBED) daemon_bit_mask=0x0000000F
[   5.699941] Service <Error> core/hle/service/service.cpp:HandleSyncRequest:97: unknown / unimplemented function '0x000900C2': port=cecd:u, cmd_buff[1]=0x0, cmd_buff[2]=0x0, cmd_buff[3]=0x5, cmd_buff[4]=0xA, cmd_buff[5]=0x0
```
The last 4 lines repeat.

**Comment by jroweboy** [comment]
@nathanhitch its fixed by https://github.com/citra-emu/citra/pull/4005 which is currently in review

**Comment by FearlessTobi** [comment]
Fixed by #4005.
